src/recyclerlistview/scrollcomponent/reactnative/ScrollComponent.js

```diff
@@ -44,7 +44,7 @@ class ScrollComponent extends React.Component {
                         {...this.props}
                         horizontal={this.props.isHorizontal}
                         onScroll={this._onScroll}
-                        onLayout={(!this._isSizeChangedCalledOnce || this.props.canChangeSize) ? this._onLayout : null}>
+                        onLayout={ !this.props.containerLayout && (!this._isSizeChangedCalledOnce || this.props.canChangeSize) ? this._onLayout : null}>
```

Comment:
Why is this needed?

Reply:
sorry,this.props.containerLayout is another feature that let user init the width and height of the container and call _onSizeChanged to finish the init at once, therefore no need to wait the onLayout call of the container

Reply:
If containerlayout has been provided it should also forcefully be applied to the scroll component?

Reply:
in my opinion，dont' need, but i'm not sure what 'forcefully be applied to the scroll component' mean